uacp/conn_test.go

```diff
@@ -6,46 +6,59 @@ package uacp
 
 import (
 	"context"
+	"net"
 	"testing"
 	"time"
 
 	"github.com/pascaldekloe/goe/verify"
 )
 
 func TestConn(t *testing.T) {
-	ep := "opc.tcp://127.0.0.1:4840/foo/bar"
-	ln, err := Listen(ep, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer ln.Close()
-
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	done := make(chan struct{})
-	acceptErr := make(chan error, 1)
-	go func() {
-		c, err := ln.Accept(ctx)
+	t.Run("server exists ", func(t *testing.T) {
+		ep := "opc.tcp://127.0.0.1:4840/foo/bar"
+		ln, err := Listen(ep, nil)
 		if err != nil {
-			acceptErr <- err
-			return
+			t.Fatal(err)
 		}
-		defer c.Close()
-		close(done)
-	}()
+		defer ln.Close()
 
-	if _, err = Dial(ctx, ep); err != nil {
-		t.Error(err)
-	}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
 
-	select {
-	case <-done:
-	case err := <-acceptErr:
-		t.Fatalf("accept fail: %v", err)
-	case <-time.After(time.Second):
-		t.Fatal("timed out")
-	}
+		done := make(chan struct{})
+		acceptErr := make(chan error, 1)
+		go func() {
+			c, err := ln.Accept(ctx)
+			if err != nil {
+				acceptErr <- err
+				return
+			}
+			defer c.Close()
+			close(done)
+		}()
+
+		if _, err = Dial(ctx, ep); err != nil {
+			t.Error(err)
+		}
+
+		select {
+		case <-done:
+		case err := <-acceptErr:
+			t.Fatalf("accept fail: %v", err)
+		case <-time.After(time.Second):
+			t.Fatal("timed out")
+		}
+	})
+
+	t.Run("Server does not exists", func(t *testing.T) {
+		ep := "opc.tcp://example.com:4840"
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		_, err := Dial(ctx, ep)
+		if err.(*net.OpError).Err.Error() != "i/o timeout" {
```

Comment:
Can we use the new `errors.Is()` here?

Reply:
the `net.OpError.Err` is a `*poll.TimeoutError`, pool is an internal package, therefore we can't compare against that.

Reply:
Ah, too bad. Would have been nice to actually see a use case for this.

Reply:
what does https://tip.golang.org/src/net/net.go?s=15864:15896#L489 return?

Reply:
it returns `true`, I guess we can use that instead.